src/stk/molecular/key_makers/molecule.py

```diff
@@ -106,9 +107,18 @@ def get_key(self, molecule):
         :class:`object`
             The key of `molecule`.
 
-        """
+        Raises
+        ------
+        :class:`ValueError`
+            If key of `molecule` is an empty string.
 
-        return self._get_key(molecule)
+        """
+        key = self._get_key(molecule)
+        if not key:
+            raise ValueError(
+                f'Key ({self._key_name}) of {molecule} is empty string'
+            )
+        return key
```

Comment:
Why is this necessary? In addition, the error message is wrong, because the return type of get_key can be any python object, not just a string. 

I also dont think the assumption that an empty string is an invalid key holds in  every possible case - which this code implies. The error raising should be done by the get_key function itself, not by the key maker, imo